scripts/TransferIndexConfig.js

```diff
@@ -43,25 +44,35 @@ class TransferIndexConfigScript extends Base {
     this.destinationIndex = destinationClient.initIndex(options.indexName);
   }
 
-  async transferIndexConfig() {
-    // Transfer settings, synonyms, and query rules
-    const settings = await this.sourceIndex.getSettings();
-    const synonyms = await this.sourceIndex.exportSynonyms();
-    const rules = await this.sourceIndex.exportRules();
-    const sOptions = {
-      forwardToReplicas: true,
-      replaceExistingSynonyms: true,
-    };
-    const rOptions = {
-      forwardToReplicas: true,
-      clearExistingRules: true,
-    };
-    await this.destinationIndex.setSettings(settings);
-    await this.destinationIndex.batchSynonyms(synonyms, sOptions);
-    await this.destinationIndex.batchRules(rules, rOptions);
+  setConfigOptions(options) {
+    this.sOptions = {};
+    this.rOptions = {};
+    if (!options.configParams) return;
+    const configParams = JSON.parse(options.configParams);
+    if (configParams.batchSynonymsParams)
+      this.sOptions = configParams.batchSynonymsParams;
+    if (configParams.batchRulesParams)
+      this.rOptions = configParams.batchRulesParams;
   }
 
-  start(program) {
+  transferIndexConfig() {
+    return new Promise(async (resolve, reject) => {
```

Comment:
you are manipulating `async/await` methods here, you don't need to create a `Promise`

Reply:
Ah, here I believe this method needs to return a promise so that on this line [here] (https://github.com/algolia/algolia-cli/pull/10/files/04fa6aacde39a726a23054c5f9962607bdbea1c5#diff-784ea65f25d224b80dd14b861bd38fb3R96) we can be sure that the next line (success console log) won't execute first.

Reply:
mmm but it will work without the new Promise, since it is and async/await context already, the thread will wait until `transferIndexConfig` is resolved (and there is a bunch of async/await in transferIndexConfig, it will wait the resolution one after one)

same as this snippet right ? 
```
(async () => {
  await waitwait5();
  console.log(3);
})();

function wait5() {
  console.log(0);
  return new Promise(resolve =>
    setTimeout(() => {
      console.log(1);
      resolve();
    }, 5000)
  );
}

async function waitwait5() {
  await wait5();
  console.log(2);
}
```
it will console.log 0,1,2,3

let me know if I missed something!!

Reply:
Ah I see, update incoming